enterprise/server/raft/filestore/filestore.go

```diff
@@ -55,10 +56,147 @@ func fileRecordSegments(r *rfpb.FileRecord) (partID string, groupID string, isol
 	return
 }
 
+type PebbleKeyVersion int
+
+const (
+	// UndefinedKeyVersion is the version of all keys in the database
+	// that have not yet been versioned.
+	UndefinedKeyVersion PebbleKeyVersion = 0
+
+	// Version1 is the first key version that includes a version in the
+	// key path, to disambiguate reading old keys.
+	Version1 = 1
+
+	// Version2 is the same as Version1, plus a change that moves the
+	// remote instance name hash to the end of the key rather than the
+	// beginning. This allows for even sampling across the keyspace,
+	// regardless of remote instance name.
+	Version2 = 2
+)
+
+type PebbleKey struct {
+	partID             string
+	groupID            string
+	isolation          string
+	remoteInstanceHash string
+	hash               string
+
+	prioritizeHashInMetadataKey bool
+}
+
+func (pmk *PebbleKey) String() string {
+	fmk, err := pmk.Bytes(UndefinedKeyVersion)
+	if err != nil {
+		return err.Error()
+	}
+	return string(fmk)
+}
+
+func (pmk *PebbleKey) Bytes(version PebbleKeyVersion) ([]byte, error) {
+	switch version {
+	case UndefinedKeyVersion:
+		filePath := filepath.Join(pmk.isolation, pmk.remoteInstanceHash, pmk.hash)
+		if pmk.isolation == "ac" {
+			filePath = filepath.Join(pmk.groupID, filePath)
+		}
+		partDir := PartitionDirectoryPrefix + pmk.partID
+		filePath = filepath.Join(partDir, filePath)
+
+		return []byte(filePath), nil
+	case Version1:
+		filePath := filepath.Join(pmk.isolation, pmk.remoteInstanceHash, pmk.hash)
+		if pmk.isolation == "ac" {
+			filePath = filepath.Join(pmk.groupID, filePath)
+		}
+		partDir := "/v1/" + PartitionDirectoryPrefix + pmk.partID
+		filePath = filepath.Join(partDir, filePath)
+		return []byte(filePath), nil
+	case Version2:
+		filePath := filepath.Join(pmk.hash, pmk.isolation, pmk.remoteInstanceHash)
+		if pmk.isolation == "ac" {
+			filePath = filepath.Join(pmk.groupID, filePath)
+		}
+		partDir := "/v3/" + PartitionDirectoryPrefix + pmk.partID
```

Comment:
v2?

Reply:
lol yeah, test caught this and I fixed it :) good 👁️